[PFS-31] Add Helm Configuration for Storage Parameters

FahadBSyed opened the pull request.

brendoncarroll commented:
This looks good from the storage side of things.  I don't think anything is missing.  I have a few concerns.
- We shouldn't double expose the GC variable.  I thought we had a way of disabling it, so people could do backups.
- This is on the pachd deployment, which is good.  How does this propagate to the storage sidecars in the worker pods.  That's where we have been wanting to configure the cache sizes.
- We need to handle defaults in one place.  I'll defer to the @pachyderm/core folks for how that's done.  But we shouldn't have defaults in the helm chart and defaults in code.  If it's in the helm chart then it should become required, or if it's missing in the helm chart, then it should be unset, so the default in the code kicks in.

FahadBSyed commented:
> This looks good from the storage side of things. I don't think anything is missing. I have a few concerns.
> 
> * We shouldn't double expose the GC variable.  I thought we had a way of disabling it, so people could do backups.  I could be misremembering.
> * This is on the pachd deployment, which is good.  How does this propagate to the storage sidecars in the worker pods.  That's where we have been wanting to configure the cache sizes?
> * We need to handle defaults in one place.  I'll defer to the @pachyderm/core folks for how that's done.  But we shouldn't have defaults in the helm chart and defaults in code.  If it's in the helm chart then it should become required, or if it's missing in the helm chart, then it should be unset, so the default in the code kicks in.

- I found the garbage collection helm parameters and updated this PR to remove the ones I added
- to propagate to pachw, I added these variables to the pachw deployment manifest as well
- defaults are probably better handled in the code to make unit and integration tests easier to set up, we could potentially comment out these helm variables in the chart like we do with other variables (Kubernetes resources). This would allow users to uncomment them when they wanted to override them.

FahadBSyed commented on etc/helm/pachyderm/values.yaml:
```diff
@@ -623,6 +623,17 @@ pachd:
+    memoryThreshold: 0
+    levelFactor: 0
+    # gcPeriod configures the garbage collection period of filesets, trackers, and temporary objects in seconds
+    gcPeriod: 60
```
Good catch, I found the variables used to expose them and removed these ones.

FahadBSyed commented on src/server/pps/server/worker_rc.go:
```diff
@@ -555,6 +561,42 @@ func (kd *kubeDriver) getStorageEnvVars(pipelineInfo *pps.PipelineInfo) []v1.Env
+			Value: strconv.FormatInt(kd.config.StorageLevelFactor, 10),
+		})
+	}
+	if kd.config.StorageCompactionMaxFanIn != 10 {
```
I am checking the defaults here because that is the established pattern, but I could also just always append these values. Would be interested in hearing peoples thoughts on what is better.
